edb/schema/pointers.py

```diff
@@ -1545,6 +1560,80 @@ def _cmd_tree_from_ast(
         return cmd
 
 
+class AlterPointerUpperCardinality(
+    referencing.ReferencedInheritingObjectCommand[Pointer_T],
+    inheriting.AlterInheritingObjectFragment[Pointer_T],
+    sd.AlterSpecialObjectField[Pointer_T],
+    PointerCommandOrFragment[Pointer_T],
+):
+    """Handler for the "cardinality" field changes."""
+
+    def _alter_begin(
```

Comment:
There's obviously a *lot* of similarity here with SetPointerType. I think a bunch of the similarity couldn't be removed without unpleasant shoehorning/excess abstraction, but I think probably the error message computation and `_set_type`/`_alter_upper_cardinality` could be consolidated?

Reply:
Yeah, some deduplication seems prudent.